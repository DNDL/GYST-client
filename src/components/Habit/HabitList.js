import React from 'react';
import PropTypes from 'prop-types';
import HabitListItem from './HabitListItem';
import styles from './habitList.css';
// import { getHabitsByUserId, getAttemptsById } from '../../selectors/habitSelectors';

// TODO: increment button
// TODO: display most recent attemps for progress visualization

function HabitList(habits) {
  console.log('in habitList', habits.habits);
  const habitElement = habits.habits.map(habit => (
    <li key={habit._id} className={styles.li}>
      <HabitListItem habit={habit}  />
    </li>
  ));

  return <ul className={styles.ul}>{habitElement}</ul>;
}

HabitList.propTypes = {
  habits: PropTypes.array
};

export default HabitList;
