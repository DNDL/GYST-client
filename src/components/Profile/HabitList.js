import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

function HabitList({ habits }) {
  const habitElement = habits.map(habit => (
    <li key={habit._id}>
      <Link to={`/habitDetail/${habit._id}`}>{habit.title}</Link>
    </li>
  ));

  return <ul>{habitElement}</ul>;
}

HabitList.propTypes = {
  habits: PropTypes.array.isRequired
};

export default HabitList;
