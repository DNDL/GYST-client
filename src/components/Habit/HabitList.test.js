import React from 'react';
import { shallow } from 'enzyme';
import HabitList from './HabitList';

describe('HabitList component', () => {
  it('renders HabitList', () => {
    const wrapper = shallow(<HabitList habits={[]}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
