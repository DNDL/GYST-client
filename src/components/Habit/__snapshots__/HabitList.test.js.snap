// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HabitList component renders HabitList 1`] = `
<Fragment>
  <section>
    <p
      className="p"
    >
      It looks like you don't have any habits yet.
      <br />
      <br />
      Click the + button below to add some!
    </p>
  </section>
</Fragment>
`;
